broadinstitute/single_cell_portal_core
# Add split view for upload options and AnnData Beta upload tab (SCP-4956)

This adds on to the uploading of a single AnnData file in place of Cluster, Expression, and Metadata files.

Videos of functionality available in these slides: https://docs.google.com/presentation/d/1QzHy48am_lqk8T3rFSL-4OmIRyaTTK_B9ozZC6HjEVs/edit#slide=id.g216c356411d_0_5

Not yet done:

- Need a back button so when users choose their upload style they can change their mind
- Need to improve look of the spinner (larger and not in corner)
- Currently need to exempt from Raw Counts by feature flag
- Backend not yet implemented
- Unit tests

To test:
- pull this branch
- start up local and navigate to a study that has no files yet
- choose the AnnData option from the split view
- fill in form info as you see fit for Expression and Cluster
- navigate to the AnnData tab and upload <this> file
- this will fail due to Raw Counts (noted above) but if you look at the terminal you will see the request show full of all the different file form info




Current shape with the new data, essentially passing everything, could prune out but wasn't sure what fully would be needed?

![Screen Shot 2023-02-27 at 4 58 38 PM](https://user-images.githubusercontent.com/54322292/221695409-1cdc6daa-9f37-439e-a30f-bcd3cdbb6dea.png)


> single_cell |   Parameters: {"study_file"=>{"name"=>"compliant_pbmc3K.h5ad", "_id"=>"63fd27241e9163ef8d0fc58a", "description"=>"", "spatial_cluster_associations"=>[""], "file_type"=>"AnnData", "upload_file_name"=>"compliant_pbmc3K.h5ad", "extra_expression_form_info_attributes"=>{"name"=>"", "_id"=>"63fd2719a7e6f745c29c0666", "status"=>"new", "description"=>"", "parse_status"=>"unparsed", "spatial_cluster_associations"=>[""], "is_spatial"=>"false", "file_type"=>"Expression Matrix", "isDirty"=>"true", "taxon_id"=>"60b917606e5ec555cec816a5"}, "expression_file_info_attributes"=>{"is_raw_counts"=>"false", "biosample_input_type"=>"Whole cell", "modality"=>"Transcriptomic: unbiased"}, "metadata_form_info_attributes"=>{"name"=>"", "_id"=>"63fd272843d2620fa90fb460", "status"=>"new", "description"=>"", "parse_status"=>"unparsed", "spatial_cluster_associations"=>[""], "file_type"=>"Metadata", "use_metadata_convention"=>"true", "isDirty"=>"true"}, "cluster_form_info_attributes"=>{"name"=>"name", "_id"=>"63fd272a1243b05ce696ca9e", "status"=>"new", "description"=>"", "parse_status"=>"unparsed", "spatial_cluster_associations"=>[""], "is_spatial"=>"false", "file_type"=>"Cluster", "isDirty"=>"true"}, "upload"=>#<ActionDispatch::Http::UploadedFile:0x00000040953ba9b8 @tempfile=#<Tempfile:/tmp/RackMultipart20230227-166-egf8zm.h5ad>, @original_filename="compliant_pbmc3K.h5ad", @content_type="application/octet-stream", @headers="Content-Disposition: form-data; name=\"study_file[upload]\"; filename=\"compliant_pbmc3K.h5ad\"\r\nContent-Type: application/octet-stream\r\n">, "parse_on_upload"=>"true"}, "study_id"=>"SCP127"}
